### ISSUE TITLE
No passthrough audio again . . .

### ISSUE THREAD

**Issue by gazzer82** [koying/SPMC#658]
I recently updated to the latest build of SPMC (I believe prior to that I was on 16.5.1) and my passthrough audio settings seems to have stopped working again.

I have my Shield hooked up to my Panasonic TV using HDMI 2.0a, the audio has fed from my TV to a ARC compatible switch that then de-embeds the audio to Optical and send it to my Sonos play bar.

Previously, with the audio settings almost identical to those recommended by Koying I was getting a nice DD stream for anything 5.1, and Stereo for all things stereo. I have for some reason had to always set number of channels to 2, and audio output device to IEC for this to work, but work it has.

However with the same settings (and actually using the recommended settings) I am weirdly only getting 5.1 (AC3) from DTS encoded videos, if I play something that is native DD I get silence.

**Settings that I have that allow 5.1 for DTS (Transcoded to DD), but silence for native DD:**

- Audio output device = Android, RAW Passthrough
- Number of channels = 2.0
- Output configuration = Best Match
- Stereo upmix = off
- Maintain original volume on downmix = on
- Boost centre channel when downmixing = 0 dB
- Resample quality = High
- Keep audio devices alive = 10 Minutes [this will prevent your AVR from going to sleep]
- Enable audio DSP processing = off
- Enable passthrough = on
- Passthrough output device = Android, RAW Passthrough

- Dolby Digital (AC3) capable receiver = on
- - Enable Dolby Digital (AC3) transcoding = on
- Dolby Digital Plus (E-AC) capable receiver = off
- DTS capable receiver = off
- TrueHD capable receiver = off
- DTS-HD capable receiver = off

**Settings that I have that outputs DTS as stereo but DD as silence:**

- Audio output device = Android, RAW Passthrough
- Number of channels = 7.1
- Output configuration = Best Match
- Stereo upmix = off
- Maintain original volume on downmix = on
- Boost centre channel when downmixing = 0 dB
- Resample quality = High
- Keep audio devices alive = 10 Minutes [this will prevent your AVR from going to sleep]
- Enable audio DSP processing = off
- Enable passthrough = on
- Passthrough output device = Android, RAW Passthrough

- Dolby Digital (AC3) capable receiver = on
- Dolby Digital Plus (E-AC) capable receiver = off
- DTS capable receiver = off
- TrueHD capable receiver = off
- DTS-HD capable receiver = off

**Settings that used to work:**

- Audio output device = Android, IEC Passthrough
- Number of channels = 2.0
- Output configuration = Best Match
- Stereo upmix = off
- Maintain original volume on downmix = on
- Boost centre channel when downmixing = 0 dB
- Resample quality = High
- Keep audio devices alive = 10 Minutes [this will prevent your AVR from going to sleep]
- Enable audio DSP processing = off
- Enable passthrough = on
- Passthrough output device = Android, IEC Passthrough

- Dolby Digital (AC3) capable receiver = on
- - Enable Dolby Digital (AC3) transcoding = on
- Dolby Digital Plus (E-AC) capable receiver = off
- DTS capable receiver = off

Any idea what's going on here, I assume it has something to do with the passed through DD stream not getting out of the shield somewhere, whereas I assume the DTS is being sent as PCM or something else?

As an aside, is there any way I can query the shield via ADP to find out what audio modes it thinks my TV is capable off? Whilst it is perfectly happy passing DD via ARC I suspect it isn't specifying that in it's EDID.

Kodi log here - http://paste.semperpax.com/pw0toattr
Shield Log here - http://paste.semperpax.com/poudwh698

The Kodi log shows me first playing a file with a DD track (Cloverfield Lane), then playing a file with a DTS track (Amelie)

The logical just shows me trying to play the first file.

What I can see in the logical is the following error, which may have something to do with this - `AudioTrack: Could not get audio output for session 33, stream type -1, usage 1, sample rate 48000, format 0x9000000, channel mask 0x3, flags 0x1`

Help appreciated as this is super confusing when everything was working fine!

Thanks

Gareth

**Comment by gazzer82** [comment]
Anyone got any ideas what may have changed to cause this not to work any more?

Thanks

Gareth

**Comment by anakin4153** [comment]
Hi,

I've roughly the same issue and don't know how to fix it.
To summarize, I have connected the shield to the TV. When I select "Enable Dolby Digital (AC3) transcoding" (basically the first configuration you provided in your post), and if I play a DTS file, the only thing I get is a kind of a white noise from the TV.

I tried all the possible configuration but without any luck ... When I play a DD file, everything is fine, but from the moment I try to convert DTS to DD, all I get is this white noise.

Please help :(

**Comment by gazzer82** [comment]
Basically the opposite of what I have then. DTS seems to convert fine but anything AC3 outputs silence. 

It used to work fine before one of the updates locked me out of the launcher and I had to reinstall. No no matter what I choose I cannot get native AC3 to work. 

It makes absolutely no sense to me as the transcoded versions play back fine. 

Unfortunately this is beyond my level of knowledge to troubleshoot future, and despite poring logs and logcats nobody seems able/willing to assist. 

Not really sure what to suggest, not sure if @koying can chime in with anything has changed in the recent builds that would cause this?

As I've previously stated, I'm pretty sure the EDID for my TV is not indicating it can output AC3, but as the DTS seems to work fine when it's converted to AC3 I don't really understand why the native AC3 doesn't. 

Can anyone help?

Thanks

Gareth

**Comment by bostjanlaba** [comment]
In my case DTS now works, DD makes silence or audio noise. In DD case the video then stutters.

**Comment by anakin4153** [comment]
Hi, 

I've the exact same issue with the video ... I began to convert all my videos to DD5.1 but it's a massive pain. Anyone for helping us ?

What TV are you using ?

**Comment by bostjanlaba** [comment]
KDL65W928C with Yamaha YSP-2700 soundbar and ARRIS IPTV STB. You're converting to DD? It's DD that I have problem with, DTS looks fine on my side.

**Comment by anakin4153** [comment]
Actually, I've a problem when converting DTS soundtrack to DD5.1 (Using AC3 Transcode) In this case, I get a white noise or silence. If I play native DD5.1 videos, it works ...

**Comment by gazzer82** [comment]
@bostjanlaba does your system support DTS or are you getting Kodi to convert it to DD? What are your audio settings?

**Comment by bostjanlaba** [comment]
my soundbar supports DD, DTS etc. I'm trying to use full passthrough. Right now I have Android RAW settings on both fields and passthrough for DTS. If I enable also DD, I get silence or static. Funny thing is like a week ago DTS and DD work working. Not anymore though.

My goal is to have SPMC passthrough all audio to my soundbar, that way I'd have a good surround in the room.

Oh, and it was working on 16.5.5. I haven't changed anything, maybe rebooted TV, but now situation is as described.

**Comment by gazzer82** [comment]
I do think something has changed recently. It 100% used to work for me with exactly the same setup as I have now. The only thing that changed was factory resetting my Shield and loading a newer version of SPMC, I think the previous version I had installed was 16.4.2, but I can't be 100% sure as the APK I installed was on a Parallels installation that got hosed. Hence loosing ADB access and having to reinstall when the auto update nukes the launcher version I had installed. 

I need to go back through the logs I think to see if I can see something different between the two playback modes (AC3 and DTS converted to AC3) to see what is different and is causing the difference!

Cheers

Gareth

**Comment by gazzer82** [comment]
Ah, I think I just worked it out. I had to disable Mediacodec (Surface) and set both output modes to Android, IEC Passthrough. More testing to follow.

**Comment by gazzer82** [comment]
Yup that seems to have fixed it, also seems to have inproved the quality of my playback, the judder I was seeing has gone away. 

Tested RAW and I still get no audio at all for DD, so it just seems to be IEC that works. 

Hopefully that helps some of you?

**Comment by anakin4153** [comment]
This is extremely interesting !
At home, if I choose IEC, I've no sound/noise, no matter what I play (DTS or DD)
If I choose RAW, then I get the DD tracks but the conversion DTS -> DD results in a noise ...

**Comment by bostjanlaba** [comment]
I've done some testing.

My results: in IEC nothing works. In RAW, things are funky but what I found out is this:
-	DTS and DTS-MA work
-	AC3 – DD works if bitrate is 448 or less. If it's 640, I get silence, video stutter and/or static noise

Does that help in any way?

Bostjan

PS: Talking about full passthrough, no conversion inside SPMC

**Comment by bostjanlaba** [comment]
Just rechecked with 10 more various movies and it appears this is the pattern in my case.

**Comment by anakin4153** [comment]
That's really super weird ... More tests here and I confirm IEC never works for me ...
RAW works only for native DD and give me noise for DTS->DD conversion ...
I now suspect my TV for being responsible, I'll try to connect the Shield directly to the Soundbar via optical link.
Did you guys try to connect the Shield directly to your TV and check if the behavior is different ?

**Comment by gazzer82** [comment]
My Shield is connected directly to my TV, I am using ARC back out of a different port to a HDMI switch to de-embed the audio to optical.

It's convoluted, but as the switcher doesn't support HDMI 2.0a it's my only option.

**Comment by gazzer82** [comment]
Now if I could just work out a way of getting DD+ out of my shield and into Optical I would b extremely happy, but as my TV doesn't support it, and none of the boxes that are out their are able to convert to DD I guess I'm stuffed. I really would quite happily pay Nvidia whatever the license cost would be to add DD+ decoding to the box as I suspect the imminent Amazon Video app will also use DD+ so I will be stuck with dull old stereo for both Amazon and Netflix whilst watching it in glorious 4K HDR, seems so very wrong :(

**Comment by CyberCopCa** [comment]
Not the same issue but the solution might address this problem

https://github.com/koying/SPMC/issues/899
